tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c

```diff
@@ -0,0 +1,470 @@
+		}
+
+		if (reset_in_progress) {
+			*err = -EFAULT;
+			return *err;
+		}
+
+		return 0;
+	}
+
+	*err = -EFAULT;
+	return *err;
+}
+
+static void fp_ram_clear(void)
+{
+	fp_storage_ak_ram_clear();
+	fp_storage_pn_ram_clear();
+	fp_storage_manager_ram_clear();
+}
+
+static void setup_fn(void)
+{
+	int err;
+
+	err = settings_validate(settings_validate_empty_cb);
+	zassert_ok(err, "Settings have not been fully deleted");
+
+	err = settings_load();
+	zassert_ok(err, "Unexpected error in settings load");
+}
+
+static int settings_get_name_cb(const char *key, size_t len, settings_read_cb read_cb, void *cb_arg,
+				void *param)
+{
+	char *name = param;
+
+	strncpy(name, key, SETTINGS_MAX_NAME_LEN);
+	return -EALREADY;
+}
+
+static int all_settings_delete(void)
+{
+	while (true) {
+		int err;
+		int ret;
+		char name[SETTINGS_MAX_NAME_LEN + 1] = "";
+
+		err = settings_load_subtree_direct(NULL, settings_get_name_cb, name);
+		if (err) {
+			return err;
+		}
+
+		ret = strcmp(name, "");
+		if (!ret) {
+			break;
+		}
+
+		err = settings_delete(name);
+		if (err) {
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+static void teardown_fn(void)
+{
+	int err;
+
+	err = all_settings_delete();
+	zassert_ok(err, "Unable to erase settings");
+
+	fp_ram_clear();
+}
+
+static void personalized_name_store(const char *name)
+{
+	int err;
+
+	err = fp_storage_pn_save(name);
+	zassert_ok(err, "Unexpected error in name save");
+}
+
+static void personalized_name_validate_loaded(const char *name)
+{
+	int err;
+	int ret;
+	char read_name[FP_STORAGE_PN_BUF_LEN];
+
+	err = fp_storage_pn_get(read_name);
+	zassert_ok(err, "Unexpected error in name get");
+	ret = strcmp(read_name, name);
+	zassert_equal(ret, 0, "Invalid Personalized Name");
+}
+
+static void fp_storage_validate_empty(void)
+{
+	int err;
+	int ret;
+	char read_name[FP_STORAGE_PN_BUF_LEN];
+	int read_count;
+	struct fp_account_key account_key_list[ACCOUNT_KEY_MAX_CNT];
+
+	err = settings_validate(fp_settings_data_validate_empty_cb);
+	zassert_ok(err, "Non-volatile Fast Pair data storage is not empty after reset");
+
+	read_count = fp_storage_ak_count();
+	zassert_equal(read_count, 0, "Invalid Account Key count");
+
+	err = fp_storage_ak_get(account_key_list, &read_count);
+	zassert_ok(err, "Unexpected error in Account Key get");
+	zassert_equal(read_count, 0, "Invalid Account Key count");
+
+	err = fp_storage_pn_get(read_name);
+	zassert_ok(err, "Unexpected error in name get");
+	ret = strcmp(read_name, "");
+	zassert_equal(ret, 0, "Expected saved name to be empty");
+}
+
+static void fp_storage_self_test_run(void)
+{
+	static const uint8_t seed = 5;
+	static const uint8_t key_count = ACCOUNT_KEY_MAX_CNT;
+	static const char *name = "fp_storage_self_test_run";
+	int err;
+
+	fp_storage_validate_empty();
+
+	cu_account_keys_generate_and_store(seed, key_count);
+	personalized_name_store(name);
+
+	cu_account_keys_validate_loaded(seed, key_count);
+	personalized_name_validate_loaded(name);
+
+	fp_ram_clear();
```

Comment:
Maybe we could check that data is unloaded at this point ?

Reply:
Also we can check it in `setup_fn`.